WEB/Src/PerformanceCollector/PerformanceCollector/QuickPulseTelemetryModule.cs

```diff
@@ -702,6 +702,18 @@ private CollectionConfigurationError[] OnUpdatedConfiguration(CollectionConfigur
         {
             // we need to preserve the current quota for each document stream that still exists in the new configuration
             CollectionConfigurationError[] errorsConfig;
+
+            lock (this.telemetryProcessorsLock)
+            {
+                foreach (var telemetryProcessor in this.TelemetryProcessors)
+                {
+                    if (telemetryProcessor is IQuickPulseTelemetryProcessor)
```

Comment:
to avoid doing the parsing twice, use something like 

```
if (telemetryProcessor is IQuickPulseTelemetryProcessor qptp)
{
      qptp.UpdateGlobalQuotas()
}
```

Reply:
ah! neat!